packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -987,6 +1036,9 @@ export class LocalFluidDataStoreContextBase extends FluidDataStoreContext {
 			props.makeLocallyVisibleFn,
 		);
 
+		// Summarizer client should not create local data stores.
+		this.identifyLocalChangeInSummarizer("DataStoreCreatedInSummarizer");
```

Comment:
Why is this not a violation? Summarizer should not have local data stores. It breaks assumptions even though we filter them out during summarize / GC, it does participate in other processes and causes issues.

Why create local data stores in summarizer if it is not intended to be used?

Reply:
Also, quoting you :-)
"can we add explicit error telemetry (no changes to behavior yet) to DDS / data store creation paths to fire if running in summarizer".
While I understand that attemp to make this data store visible will be caught by the op submission path (attach op) but the existence of local data stores itself causes problems. For example, we are seeing instances where a local data store is created but its realization (hasInitialized) fails so that the data store is never attached but it creates issues because it has a summarizer node. We could exclude creating summarizer nodes for local data stores but in the interest of fail fast, we should just fail here.